code/modules/uplink/uplink_items.dm

```diff
@@ -142,6 +142,11 @@
 		spawned_item = spawn_path
 	if(refundable)
 		spawned_item.AddElement(/datum/element/uplink_reimburse, (refund_amount ? refund_amount : cost))
+	if(category != /datum/uplink_category/stealthy_tools) // Things that shouldn't be detectable as contraband on the station.
+		ADD_TRAIT(spawned_item, TRAIT_CONTRABAND, TRAIT_GENERIC)
+		if(spawned_item.contents)
+			for(var/obj/contained as anything in spawned_item.contents)
+				ADD_TRAIT(contained, TRAIT_CONTRABAND, TRAIT_GENERIC)
```

Comment:
You may see it fit to put this in `spawn_item_for_generic_use` (or you may not, up to you)

Also doesn't this technically mean uplink items created via the surplus crate or obtained with other methods (IE, found in space loot) are now have a literal antag tell, as only items purchased from the uplink will be contraband?

Reply:
That would be true I suppose, it might be worth perhaps adding a mapping helper that applies additional contraband traits to items on a turf to avoid the disparity.

Reply:
That seems like a lot of added complexity for mappers